shanoir-uploader/src/main/java/org/shanoir/uploader/action/ImportFromCsvActionListener.java

```diff
@@ -0,0 +1,412 @@
+import org.shanoir.uploader.model.rest.importer.ImportJob;
+import org.shanoir.uploader.model.rest.importer.Instance;
+import org.shanoir.uploader.nominativeData.NominativeDataUploadJob;
+import org.shanoir.uploader.nominativeData.NominativeDataUploadJobManager;
+import org.shanoir.uploader.service.rest.ShanoirUploaderServiceClientNG;
+import org.shanoir.uploader.utils.Util;
+import org.shanoir.util.ShanoirUtil;
+
+public class ImportFromCsvActionListener implements ActionListener {
+
+	ImportFromCSVWindow importFromCSVWindow;
+	IDicomServerClient dicomServerClient;
+	File shanoirUploaderFolder;
+
+	List<CsvImport> csvImports;
+	ShanoirUploaderServiceClientNG shanoirUploaderServiceClientNG;
+
+	private static Logger logger = Logger.getLogger(ImportFromCsvActionListener.class);
+
+	public ImportFromCsvActionListener(ImportFromCSVWindow importFromCSVWindow, IDicomServerClient dicomServerClient, File shanoirUploaderFolder, ShanoirUploaderServiceClientNG shanoirUploaderServiceClientNG) {
+		this.importFromCSVWindow = importFromCSVWindow;
+		this.dicomServerClient = dicomServerClient;
+		this.shanoirUploaderFolder = shanoirUploaderFolder;
+		this.shanoirUploaderServiceClientNG = shanoirUploaderServiceClientNG;
+	}
+
+	@Override
+	public void actionPerformed(ActionEvent e) {
+		// Iterate over import to import them one by one
+		Set<Long> idList = new HashSet<>();
+		Map<String, ArrayList<StudyCard>> studyCardsByStudy = new HashMap<>();
+
+		for (CsvImport importTodo : this.csvImports) {
+			idList.add(Long.valueOf(importTodo.getStudyId()));
+			studyCardsByStudy.put(importTodo.getStudyId(), new ArrayList<StudyCard>());
+		}
+		
+		List<org.shanoir.uploader.model.rest.Study> studies = shanoirUploaderServiceClientNG.findStudiesNamesAndCenters();
+
+		IdList idealist = new IdList();
+		idealist.setIdList(new ArrayList<>(idList));
+		List<StudyCard> studyCards = shanoirUploaderServiceClientNG.findStudyCardsByStudyIds(idealist);
+		studyCards.stream().forEach( element -> {
+			try {
+				AcquisitionEquipment acquisitionEquipment = shanoirUploaderServiceClientNG.findAcquisitionEquipmentById(element.getAcquisitionEquipmentId());
+				element.setAcquisitionEquipment(acquisitionEquipment);
+				element.setCenterId(acquisitionEquipment.getCenter().getId());
+			} catch (Exception e1) {
+				//TODO: error
+				e1.printStackTrace();
+			}
+			studyCardsByStudy.get(element.getStudyId().toString()).add(element);
+		});
+
+		for (CsvImport importTodo : this.csvImports) {
+			importData(importTodo, studyCardsByStudy.get(importTodo.getStudyId()), studies.stream().filter(element -> element.getId().toString().equals(importTodo.getStudyId())).findFirst().get());
+		}
+	}
+
+	/**
+	 * Loads data to shanoir NG
+	 * @param csvImport the import
+	 * @param studyCardsByStudy the list of study
+	 * @param study2
+	 * @return
+	 */
+	private void importData(CsvImport csvImport, List<StudyCard> studyCardsByStudy, org.shanoir.uploader.model.rest.Study study2) {
```

Comment:
Hi Jean-Come, I am fully aware that this is a draft. Please let me know if you prefer that I keep my comments for a later time. I think, this method contains quite a lot of duplicate, but necessary code. Maybe it would be good to refactor some parts for commonly used components to avoid duplication? With kind regards, Michael

Reply:
I need to:
- Refactor to avoid code duplication
- Clean the code and do a proper import with the CSV input values (+ check values at the entry)
- Better / mix display for the imports
- Test with multiple imports and avoir to parallelize too much import (I have to think of it)

I was just creating the pull request to get an overwiew